pkg/gotohelm/transpiler.go

```diff
@@ -858,15 +869,28 @@ func (t *Transpiler) transpileCallExpr(n *ast.CallExpr) Node {
 	}
 
 	// Method call.
-	if callee.Type().(*types.Signature).Recv() != nil {
+	if r := callee.Type().(*types.Signature).Recv(); r != nil {
 		if len(args) != 0 {
 			panic(&Unsupported{Fset: t.Fset, Node: n, Msg: "method calls with arguments are not implemented"})
 		}
-		// Method calls come in as a "top level" CallExpr where .Fun is the
-		// selector up to that call. IE all of `Foo.Bar.Baz()` will be "within"
-		// the CallExpr. CallExpr.Fun will contain Foo.Bar.Baz. In the case of
-		// zero argument methods, text/template will automatically call them.
-		return t.transpileExpr(n.Fun)
+
+		if s, ok := n.Fun.(*ast.SelectorExpr); ok {
+			str := r.Type().String()
+			if s.Sel.Name == "AsMap" && str == "github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette.Values" {
+				// In helm the AsMap method could be used for helm Values
+				return t.transpileExpr(n.Fun)
+			}
+		}
+
+		if typeName, ok := r.Type().(*types.Pointer); ok {
+			if baseTypeName, ok := typeName.Elem().(*types.Named); ok {
+				return &Call{
+					FuncName:  fmt.Sprintf("%s.%s.%s", t.Package.Name, baseTypeName.Obj().Name(), callee.Name()),
+					Arguments: []Node{t.transpileExpr(n.Fun.(*ast.SelectorExpr).X)},
```

Comment:
Can you copy the original comment about method calls to be above this line? 

Here's a slightly more clear version of it:

```
// Method calls come in as a "top level" CallExpr where .Fun is the
// selector up to that call. e.g. `Foo.Bar.Baz()` will be a `CallExpr`.
// It's `.Fun` is a `SelectorExpr` where `.X` is `Foo.Bar`, the receiver,
// and `.Sel` is `Baz`, the method name.
```